pkg/agent/agent.go

```diff
@@ -112,6 +123,41 @@ func (a *Agent) shutdown() {
 	}
 }
 
+func (a *Agent) setupProfiling() {
+	if a.c.ProfilingEnabled {
+		if runtime.MemProfileRate == 0 {
+			a.c.Log.Warn("Memory profiles are disabled")
+		}
+		if a.c.ProfilingPort > 0 {
+			grpc.EnableTracing = true
+			go func() {
+				port := strconv.Itoa(a.c.ProfilingPort)
```

Comment:
might be able to accomplish this with fmt.Sprintf, which can convert the integer as well as form the localhost: string in one step

Reply:
:+1: